### PR TITLE
fix: DE42230: Add follow rel path to list-item-detailed

### PR DESCRIPTION
**Pull request by surags**
https://rally1.rallydev.com/#/357251704080d/custom/367300408400?detail=%2Fdefect%2F486936639864&fdp=true&fdp=true?fdp=true

### CODE REVIEW

File: components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js [hunk 1]
```diff
@@ -374,24 +357,47 @@ class ActivityListItemDetailed extends ListItemLinkMixin(SkeletonMixin(EntityMix
...
+					this._activity = foundEntity;
+
+					const link = (
+						this._activityProperties
+						&& this._activityProperties.linkRel
+						&& this._activity.getLinkByRel(this._activityProperties.linkRel)
+					) || this._activity.getLinkByRel('alternate');
 
-				break;
+					this.actionHref = (this._started && (this.evaluateAllHref || (link && link.href))) || null;
```
Comment (bearfriend) [inline components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js:373]: Only thing is there is no `evaluateAllHref` in the detailed view, so:
`(this._started && link && link.href) || null;` I think?

File: components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js [hunk 2]
```diff
@@ -374,24 +357,47 @@ class ActivityListItemDetailed extends ListItemLinkMixin(SkeletonMixin(EntityMix
...
-					|| {}).href;
-				if (source) {
-					await fetchEntity(source, this.token)
-						.then((sirenEntity) => {
-							if (sirenEntity) {
-								this._activity = sirenEntity;
-							}
-						});
-				}
+				const relList = [].concat(this._activityProperties.rel);
```
Comment (mbardakovD2L) [inline components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js:360]: probably a dumb question but I feel obligated to ask - do we know for sure that the rels we need to dig through are going to be in the right order?

File: components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js (hunk 1 above)
Comment (surags) [reply components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js]: 🤦‍♂️

File: components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js (hunk 2 above)
Comment (surags) [reply components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js:360]: The relList is defined [here](https://github.com/BrightspaceHypermediaComponents/activities/blob/854bbe1ce5a77387633dd0ef4d9120d302212348/components/d2l-work-to-do/env/index.js#L47) so the order will always be known.

File: components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js [hunk 3]
```diff
@@ -374,24 +357,47 @@ class ActivityListItemDetailed extends ListItemLinkMixin(SkeletonMixin(EntityMix
...
+
+				const foundEntity = await this._followRelPath(relList, entity);
+
+				if (foundEntity) {
+					this._activity = foundEntity;
+
+					const link = (
```
Comment (JasperAtD2L) [inline components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js:367]: Should leave this as a getter/setter, to be consistent with the basic component. These 2 components really should've extended from each other.

Comment (JasperAtD2L) [reply components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js:367]: Oh I see you are basing of Danny's PR: https://github.com/BrightspaceHypermediaComponents/activities/pull/1443

Comment (bearfriend) [reply components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js:367]: Yeah working on getting that merged right now so we can clear this chain of dependent PRs.

File: components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js [hunk 4]
```diff
@@ -374,24 +357,47 @@ class ActivityListItemDetailed extends ListItemLinkMixin(SkeletonMixin(EntityMix
...
+						this._activityProperties
+						&& this._activityProperties.linkRel
+						&& this._activity.getLinkByRel(this._activityProperties.linkRel)
+					) || this._activity.getLinkByRel('alternate');
 
-				break;
```
Comment (JasperAtD2L) [inline components/d2l-work-to-do/d2l-work-to-do-activity-list-item-detailed.js:390]: Keep the break, as we don't have to go through the rest of the allowed list.

**Review by JasperAtD2L — APPROVED** [review]

**Pull request merged**